src/parseTools.js

```diff
@@ -8,7 +8,7 @@
  * Tests live in test/other/test_parseTools.js.
  */
 
-const FOUR_GB = 4 * 1024 * 1024 * 1024;
+globalThis.FOUR_GB = 4 * 1024 * 1024 * 1024;
```

Comment:
Perhaps we should look at why this change was needed?   Presumably it means that some things that were previously available in preprocessor statements might no longer be?   I wonder how many such cases there are?

Reply:
The only things that should / would become unavailable now are local variables in scope, since now I'm using the safer global eval, as every other place except the preprocessor, already used. All the settings and helper functions are already assigned to the global object so continue working as expected.

This file declared two variables - FOUR_GB and FLOAT_TYPES - and, from my quick grep, only FOUR_GB is actualy used in libraries. It worked before because it happened to be in the same scope as `eval`, but instead, like other settings, it should be set on the global object to be consistently available everywhere else. Which is what I did here.

In theory it's possible that someone used local variables from `preprocessor` function too (e.g. `i` or `line` or `filenameHint`) but that would be extremely fragile and I really hope nobody did that...

At least the tests pass now, but with share-everything eval it used before, you can never be sure what external users relied on 🤷‍♂️ Mandatory https://xkcd.com/1172/ as always applies.

Reply:
Makes sense.  lgtm